From 41472f4d98d165465bc9aa9bc399911b7bc3c50e Mon Sep 17 00:00:00 2001
From: Baptiste Assmann <bedis9@gmail.com>
Date: Fri, 8 May 2015 23:34:06 +0200
Subject: [PATCH 06/28] MINOR: server: Macro definition for server-state

Introduces a few new macros used by server state save and application accros reloads:
- currently used state server file format version
- currently used state server file header fields
- MIN and MAX value for version number
- maximum number of fields that could be found in a server-state file
- an arbitrary state-file max line length
---
 include/types/server.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/include/types/server.h b/include/types/server.h
index 40b8308..abd1130 100644
--- a/include/types/server.h
+++ b/include/types/server.h
@@ -82,6 +82,16 @@ enum srv_admin {
 	SRV_ADMF_DRAIN     = 0x18,        /* mask to check if any drain flag is present */
 };
 
+/* server-state-file version */
+#define SRV_STATE_FILE_VERSION 1
+#define SRV_STATE_FILE_VERSION_MIN 1
+#define SRV_STATE_FILE_VERSION_MAX 1
+#define SRV_STATE_FILE_FIELD_NAMES "be_id be_name srv_id srv_name srv_addr srv_op_state srv_admin_state srv_uweight srv_iweight srv_time_since_last_change srv_check_status srv_check_result srv_check_health srv_check_state srv_agent_state bk_f_forced_id srv_f_forced_id"
+#define SRV_STATE_FILE_MAX_FIELDS 18
+#define SRV_STATE_FILE_NB_FIELDS_VERSION_1 18
+#define SRV_STATE_LINE_MAXLEN 512
+
+
 /* server flags */
 #define SRV_F_BACKUP       0x0001        /* this server is a backup server */
 #define SRV_F_MAPPORTS     0x0002        /* this server uses mapped ports */
-- 
2.4.9

